src/foam/nanos/auth/login/SignUp.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license
+ * Copyright 2023 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth.login',
+  name: 'SignUp',
+
+  requires: [
+    'foam.nanos.auth.User'
+  ],
+
+  messages: [
+    { name: 'EMAIL_ERR', message: 'Valid email required' },
+    { name: 'PASSWORD_ERR', message: 'Password should be at least 10 characters' },
+    { name: 'USERNAME_EMPTY_ERR', message: 'Username required' },
+    { name: 'USERNAME_ERR', message: 'Valid username required' },
+    { name: 'USERNAME_AVAILABILITY_ERR', message: 'This username is taken. Please try another.' },
+    { name: 'EMAIL_AVAILABILITY_ERR', message: 'This email is already in use. Please sign in or use a different email' },
+    { name: 'WEAK_PASSWORD_ERR', message: 'Password is weak' }
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'username',
+      label: 'Username',
+      placeholder: 'example123',
+      view: function(_, X) {
+        return {
+          class: 'foam.u2.view.UserPropertyAvailabilityView',
+          icon: 'images/checkmark-small-green.svg',
+          isAvailable$: X.data.usernameAvailable$,
+          inputValidation: X.data.User.USER_NAME_MATCHER
+        };
+      },
+      validationPredicates: [
+        {
+          args: ['username'],
+          query: 'username.len>0',
+          errorMessage: 'USERNAME_EMPTY_ERR'
+        },
+        {
+          args: ['usernameAvailable'],
+          query: 'usernameAvailable==true',
+          errorMessage: 'USERNAME_AVAILABILITY_ERR'
+        }
+      ]
+    },
+    {
```

Comment:
Has been updated in https://github.com/kgrgreer/foam3/commit/0cb137d38897cf1ecc77db23eb9e0e842516fe38#diff-0026cfc7e2391117957f00f1cc83489c816b140fa9533d11f7b475b2cf7d7ad7